src/components/Organisms/TodoCard/index.jsx

```diff
@@ -0,0 +1,98 @@
+import React, { useRef, useEffect, useState } from "react";
+import styled from "styled-components";
+import AddTaskButton from "../../Atoms/AddTaskButton/index";
+import Task from "../../Molecules/Task/index";
+import COLOR from "../../../variables/color";
+
+const TodoCard = () => {
+  const [todoTask, setTodoTask] = useState([]);
+  const [taskAddOREdit, setAddOREdit] = useState(false);
+
+  let todoTaskList = [];
+  let todoStateTask = [];
+
+  const handleTaskComplete = (id) => {
+    let completedTask = todoTask.map((task) => {
+      if (task.id === id) {
+        task.state = "DONE";
+      }
+      return task;
+    });
+    setTodoTask(completedTask);
+  };
+
+  const handleEditComplete = (id, taskName) => {
+    let edittedTask = [];
+    if (taskName === "") {
+      edittedTask = todoTask.filter((task) => {
+        return task.id !== id;
+      });
+    } else {
+      edittedTask = todoTask.map((task) => {
+        if (task.id === id) {
+          task.name = taskName;
+        }
+        return task;
+      });
+    }
+    setTodoTask(edittedTask);
+    setAddOREdit(false);
+  };
+
+  // task.stateがTODOのものだけ抽出する
+  todoStateTask = todoTask.filter((task) => {
+    return task.state === "TODO";
+  });
+
+  todoTaskList = todoStateTask.map((task, index) => {
+    // addTask()が実行されたとき、todoTaskの一番後ろのコンポーネントだけにdefaultIsEditingを渡す
+    return (
+      <StyledTodoListItem key={index}>
+        <Task
+          key={task.id}
+          defalutValue={task.name}
+          defalutIsEditing={
+            taskAddOREdit ? task.id === todoTask.length - 1 : false
+          }
+          onEditComplete={(taskName) => handleEditComplete(task.id, taskName)}
+          onTaskComplete={() => handleTaskComplete(task.id)}
+        />
+      </StyledTodoListItem>
+    );
+  });
+
+  const addTask = () => {
+    const idNum = todoTask.length;
+    setTodoTask((todoTask) => [
+      ...todoTask,
+      { id: idNum, name: "", state: "TODO" },
+    ]);
+    setAddOREdit(true);
+  };
+
+  return (
+    <StyledTodoCard>
+      <AddTaskButton onClick={addTask} />
+      <StyledTodoList>{todoTaskList}</StyledTodoList>
+    </StyledTodoCard>

```

Comment:
```suggestion
    <StyledTodoCard>
      <AddTaskButton onClick={addTask} />
      <StyledTodoList>
      　{ todoStateTask.map((task, index) =>  (
      <StyledTodoListItem key={index}>
        <Task
          key={task.id}
          defalutValue={task.name}
          defalutIsEditing={
            taskAddOREdit ? task.id === todoTask.length - 1 : false
          }
          onEditComplete={(taskName) => handleEditComplete(task.id, taskName)}
          onTaskComplete={() => handleTaskComplete(task.id)}
        />
      </StyledTodoListItem>
      )}
      </StyledTodoList>
    </StyledTodoCard>
```
コンポーネント内でJSXの記述が2つに散っているのが気になるので、このように一つにまとめてもいいと思います。
（ここは感性の問題な気もするので修正しなくても大丈夫です。）